From dfd7195a68388d945f8df06a4d460cad2459fe68 Mon Sep 17 00:00:00 2001
From: yan cui <ccuiyyan@gmail.com>
Date: Fri, 29 Jul 2016 23:25:44 +0000
Subject: [PATCH] bug fix

---
 sys/kern/kern_sig.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/sys/kern/kern_sig.c b/sys/kern/kern_sig.c
index 3d9d68e..f3681d3 100644
--- a/sys/kern/kern_sig.c
+++ b/sys/kern/kern_sig.c
@@ -547,6 +547,7 @@ siggetinfo(sigpend_t *sp, ksiginfo_t *out, int signo)
 			memcpy(out, ksi, sizeof(*out));
 			out->ksi_flags &= ~(KSI_FROMPOOL | KSI_QUEUED);
 		}
+		ksiginfo_free(ksi);
 	}
 	return ret;
 
-- 
2.6.4

