diff -ruNa c/posixtestsuite/LDFLAGS patched/LDFLAGS
--- c/posixtestsuite/LDFLAGS	2016-08-18 15:29:52.000000000 -0700
+++ patched/LDFLAGS	2016-08-18 15:30:54.000000000 -0700
@@ -9,6 +9,7 @@
 #
 #Recommended flags:
 #-D_XOPEN_SOURCE=600 -lpthread -lrt -lm
+-lpthread -lrt -lm
 #
 # For use with Linux, you may try the following flags to
 # allow for the NPTL-specific compilation (used in some test cases)
diff -ruNa c/posixtestsuite/Makefile patched/Makefile
--- c/posixtestsuite/Makefile	2016-08-18 15:29:52.000000000 -0700
+++ patched/Makefile	2016-08-18 15:31:22.000000000 -0700
@@ -54,7 +54,7 @@
 tests-pretty:
 	$(MAKE) all | column -t -s:
 
-CFLAGS = -g -O2 -Wall -Werror -D_POSIX_C_SOURCE=200112L
+CFLAGS = -g -O2
 
 # add -std=c99, -std=gnu99 if compiler supports it (gcc-2.95.3 does not).
 check_gcc = $(shell if $(CC) $(1) -S -o /dev/null -xc /dev/null > /dev/null 2>&1; then echo "$(1)"; else echo "$(2)"; fi)
diff -ruNa c/posixtestsuite/stress/mqueues/Makefile patched/stress/mqueues/Makefile
--- c/posixtestsuite/stress/mqueues/Makefile	2016-08-18 15:29:52.000000000 -0700
+++ patched/stress/mqueues/Makefile	2016-08-18 15:32:02.000000000 -0700
@@ -4,7 +4,7 @@
 endif
 
 INCLUDE = -I../../include
-LIB=-lmqueue
+LIB=-lrt
 
 CC=gcc
 CFLAGS=-Wall -O2 -g -I$(POSIX_DIR_INC) -L$(POSIX_DIR_LIB) -lpthread
diff -ruNa c/posixtestsuite/stress/semaphores/Makefile patched/stress/semaphores/Makefile
--- c/posixtestsuite/stress/semaphores/Makefile	2016-08-18 15:29:52.000000000 -0700
+++ patched/stress/semaphores/Makefile	2016-08-18 15:32:33.000000000 -0700
@@ -3,7 +3,6 @@
 endif
 
 INCLUDE = -I../../include
-LIB=-lposix1b
 
 CC=gcc
 CFLAGS=-Wall -O2 -g -I$(POSIX_DIR_INC) -L$(POSIX_DIR_LIB)
