From d28b50871d63391c12da31a415f8dc3efa388167 Mon Sep 17 00:00:00 2001
From: yan cui <ccuiyyan@gmail.com>
Date: Sun, 15 May 2016 03:00:47 +0000
Subject: [PATCH] expose SIGRTMIN and SIGRTMAX

---
 sys/sys/signal.h | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/sys/sys/signal.h b/sys/sys/signal.h
index bec4aa1..92a936f 100644
--- a/sys/sys/signal.h
+++ b/sys/sys/signal.h
@@ -82,10 +82,8 @@
 #define	SIGUSR1		30	/* user defined signal 1 */
 #define	SIGUSR2		31	/* user defined signal 2 */
 #define	SIGPWR		32	/* power fail/restart (not reset when caught) */
-#ifdef _KERNEL
-#define	SIGRTMIN	33	/* Kernel only; not exposed to userland yet */
-#define	SIGRTMAX	63	/* Kernel only; not exposed to userland yet */
-#endif
+#define	SIGRTMIN	33
+#define	SIGRTMAX	63
 
 #ifndef _KERNEL
 #include <sys/cdefs.h>
-- 
2.6.4

