From 18dde18546744a5da6e098893337f9af9b7a6649 Mon Sep 17 00:00:00 2001
From: yan cui <ccuiyyan@gmail.com>
Date: Wed, 18 May 2016 02:44:14 +0000
Subject: [PATCH] add bsd_signal

---
 sys/sys/signal.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/sys/sys/signal.h b/sys/sys/signal.h
index 92a936f..47e3cd7 100644
--- a/sys/sys/signal.h
+++ b/sys/sys/signal.h
@@ -233,6 +233,7 @@ struct	sigevent {
  */
 __BEGIN_DECLS
 void	(*signal(int, void (*)(int)))(int);
+#define	bsd_signal signal
 #if (_POSIX_C_SOURCE - 0) >= 200112L || defined(_NETBSD_SOURCE)
 int	sigqueue(pid_t, int, const union sigval);
 #endif
-- 
2.6.4

